enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,59 @@
+# Enhancement of cluster deletion 
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+Currently, when we remove a cluster from the Hub, the `managedClusterAddons` and `manifestWorks` in the cluster namespace need to be deleted manually, and cannot ensure that the resources deployed on the Spoke cluster can be cleaned up. This proposal is to define clearly what happened when a cluster is deleted.
+
+## Motivation
+
+### Goals
+
+- Define which resources related the deleted cluster should be cleaned up by OCM.
+- Clarify the order in which these resources are deleted.
+
+### Non-Goals
+
+- Clean up customized resources automatically.
+
+## Proposal
+
+### Which resources should be cleaned up when a cluster is removed from the Hub.
+
+On the Hub cluster
+
+1. The `managedCluster` should be deleted.
+2. The `managedClusterAddons` and `manifestWorks` in the cluster namespace should be cleaned up by OCM.
+3. The other resources in the cluster namespace except `managedClusterAddons` and `manifestWorks` should be cleaned up by their owners.
+4. The cluster namespace should be deleted too.
+   
+On the Spoke cluster
+
+1. The resources deployed by the `manifestWorks` should be cleaned up by OCM. 
+2. The resources deployed by other owners can be cleaned up optionally by their owners like the subscriptions deployed by application addon.
+3. The `Klusterlet` CR should be deleted.
+
+### The behaviors when a cluster is removed from the Hub.
+
+1. The first step to remove a cluster is to delete the `managedCluster` on the Hub.
+2. Delete the `managedClusterAddons` and `manifestWorks` in the cluster namespace.
+3. Delete the `roleBindings` of registration-agent and work-agent in the cluster namespace after all of `managedClusterAddons` and `manifestWorks` are deleted.
+4. Delete the cluster namespace.
```

Comment:
we do not delete cluster namespace today. Do we need to do this? Or we should by default delete ns and allow user to choose not to delete ns?

Reply:
considering to add a `deleteOption` filed in the spec of managedCluster.
Set `deleteAll` to delete the cluster namespace after the managedCluster is deleted.
Otherwise, the cluster namespace will not be deleted by default.

and define an annotation `open-cluster-management.io/delete-by-user`. The managedClusterAddons and manifestWorks with this annotation will not be deleted by OCM during the deletion process.

Reply:
will not delete the cluster ns with the label `cluster.open-cluster-management.io/keep-upon-cluster-deletion`